src/Illuminate/Macroable/Traits/Macroable.php

```diff
@@ -31,18 +31,28 @@ public static function macro($name, $macro)
     /**
      * Mix another object into the class.
      *
-     * @param  object  $mixin
+     * @param  mixed  $mixin
      * @param  bool  $replace
      * @return void
      *
      * @throws \ReflectionException
      */
     public static function mixin($mixin, $replace = true)
     {
+        if (is_iterable($mixin)) {
```

Comment:
Doesn't this break things for objects that were already iterable?

Reply:
I added a test for arrays, but if you could add one that will make it fail that would be great. 👍

Reply:
this is still breaking

Reply:
object and iterable are not disjoint types

Reply:
I believe need to allow object|array and checking on is_array

Reply:
using `object|array` and `is_array` would remove the support for `Collections` but since that use-case would be minimal  I guess just going with `is_array` would be the smarter choice. Thoughts @GrahamCampbell?